core/model-api/src/main/java/org/eclipse/rdf4j/model/base/AbstractValueFactory.java

```diff
@@ -0,0 +1,280 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Eclipse RDF4J contributors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Distribution License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ ******************************************************************************/
+
+package org.eclipse.rdf4j.model.base;
+
+import static org.eclipse.rdf4j.model.base.AbstractLiteral.reserved;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.atomic.AtomicLong;
+
+import javax.xml.datatype.XMLGregorianCalendar;
+
+import org.eclipse.rdf4j.model.BNode;
+import org.eclipse.rdf4j.model.IRI;
+import org.eclipse.rdf4j.model.Literal;
+import org.eclipse.rdf4j.model.Resource;
+import org.eclipse.rdf4j.model.Statement;
+import org.eclipse.rdf4j.model.Triple;
+import org.eclipse.rdf4j.model.Value;
+import org.eclipse.rdf4j.model.ValueFactory;
+import org.eclipse.rdf4j.model.base.AbstractBNode.GenericBNode;
+import org.eclipse.rdf4j.model.base.AbstractIRI.GenericIRI;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.BooleanLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.CalendarLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.DecimalLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.IntegerLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.NumberLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.TaggedLiteral;
+import org.eclipse.rdf4j.model.base.AbstractLiteral.TypedLiteral;
+import org.eclipse.rdf4j.model.base.AbstractStatement.GenericStatement;
+import org.eclipse.rdf4j.model.base.AbstractTriple.GenericTriple;
+
+/**
+ * Base class for {@link ValueFactory}, offering common functionality.
+ *
+ * @author Alessandro Bollini
+ * @since 3.5.0
+ */
+@SuppressWarnings("UseOfObsoleteDateTimeApi")
+public abstract class AbstractValueFactory implements ValueFactory {
+
+	private static final Literal TRUE = new BooleanLiteral(true);
+	private static final Literal FALSE = new BooleanLiteral(false);
+
+	////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	private final AtomicLong nodeID = new AtomicLong(ThreadLocalRandom.current().nextLong());
+
+	@Override
+	public BNode createBNode() {
+		return new GenericBNode(Long.toHexString(Math.abs(nodeID.getAndIncrement())));
+	}
+
+	@Override
+	public BNode createBNode(String nodeID) {
+
+		if (nodeID == null) {
+			throw new NullPointerException("null nodeID");
+		}
```

Comment:
`Objects.requireNonNull(nodeID);`?

Reply:
No problem migrating to the compact form, if that is the project standard; however, in the context of argument checking, I personally find the extended form more explicit and more consistent when accompanied by other checks as in:

```java
if (iri == null) {
	throw new NullPointerException("null iri");
}

if (iri.indexOf(':') < 0) {
	throw new IllegalArgumentException("missing colon in absolute IRI");
}
```

Reply:
Migrated to `Objects.requireNonNull()`.

Reply:
I don't think we have a consistent standard for this throughout the project - I tend to prefer the more compact form myself but there's some room for personal preference here :)